spec.md

```diff
@@ -772,6 +772,155 @@ A function or method may be referenced in a way that creates a closure out of it
 
 If so, the `...` MUST NOT include any whitespace before or after. That is, the correct format is `foo(...)`.
 
+
+### 4.9 Property Hooks
+
+Object properties may also include hooks, which have a number of syntactic options.
+
+When using the long form of hooks:
+
+* The opening brace MUST be on the same line as the property.
+* The opening brace MUST be separated from the property name or its default value by a single space.
+* The closing brace MUST be on its own line, and have no comment following it.
+* The entire body of the hook definition MUST be indented one level.
+* The body of each hook MUST be indented one level.
+* If multiple hooks are declared, they MUST be separated by at least a single line break.  They
+  MAY be separated by an additional blank line to aid readability.
+
+For example:
+
+```php
+class Example
+{
+    public string $newName = 'Me' {
+        set {
+            if (strlen($value) < 3) {
+                throw new \Exception('Too short');
+            }
+            $this->newName = ucfirst($value);
+        }
+    }
+
+    public string $department {
+        get {
+            return $this->values[__PROPERTY__];
+        }
+        set {
+            $this->values[__PROPERTY__] = $value;
+        }
+    }
+    // or
+    public string $department {
+        get {
+            return $this->values[__PROPERTY__];
+        }
+
+        set {
+            $this->values[__PROPERTY__] = $value;
+        }
+    }
+}
+```
+
+Property hooks also support multiple short-hook variations.
+
+For a `set` hook, if the argument name and type do not need to be redefined, then they MAY be omitted.
+
+If a hook consists of a single expression, then PHP allows it to be shortened using `=>`.  In that case:
+
+* There MUST be a single space on either side of the `=>` symbol.
+* The body MUST be on the same line as the hook declaration, unless it gets prohibitively long.  If it gets prohibitively long, the developer SHOULD consider not using the short-hook syntax.
```

Comment:
This entire line is not sufficiently actionable. Combining a MUST with an unless doesn't make sense.

We already have a character limit per line, perhaps we can refer to that here?

> The body MUST be on the same line as the hook declaration and MUST NOT be wrapped. If wrapping is necessary to keep the line within the line length or desired for other reasons the short-hook syntax MUST NOT be used.

Reply:
Can you make an edit for that so I can merge it?